evap/results/tools.py

```diff
@@ -471,49 +471,52 @@ def textanswers_visible_to(contribution):
     user: UserProfile,
     represented_users: list[UserProfile],
     textanswer: TextAnswer,
-    view: str,
+    view_general_text: str,
+    view_contributor_results: str,
 ) -> bool:
     assert textanswer.review_decision in [TextAnswer.ReviewDecision.PRIVATE, TextAnswer.ReviewDecision.PUBLIC]
     contributor = textanswer.contribution.contributor
 
-    if view == "public":
-        return False
-
-    if view == "export":
-        if textanswer.is_private:
-            return False
-        if not textanswer.contribution.is_general and contributor != user:
-            return False
-    elif user.is_reviewer:
-        return True
-
-    if textanswer.is_private:
-        return contributor == user
-
     # NOTE: when changing this behavior, make sure all changes are also reflected in results.tools.textanswers_visible_to
     # and in results.tests.test_tools.TestTextAnswerVisibilityInfo
+    # überall wo das alte benutzt wurde anpassen
     if textanswer.is_public:
-        # users can see textanswers if the contributor is one of their represented users (which includes the user itself)
-        if contributor in represented_users:
-            return True
-        # users can see text answers from general contributions if one of their represented users has text answer
-        # visibility GENERAL_TEXTANSWERS for the evaluation
-        if (
-            textanswer.contribution.is_general
-            and textanswer.contribution.evaluation.contributions.filter(
-                contributor__in=represented_users,
-                textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
-            ).exists()
-        ):
-            return True
-        # the people responsible for a course can see all general text answers for all its evaluations
-        if textanswer.contribution.is_general and any(
-            user in represented_users for user in textanswer.contribution.evaluation.course.responsibles.all()
-        ):
-            return True
+        if textanswer.contribution.is_general:
+            if view_general_text == "full":  # filters all contributions
+                # reviewer can see everything
+                if user.is_reviewer:
+                    return True
 
+                # user can see general textanswer if represented user can see it
+                if textanswer.contribution.evaluation.contributions.filter(
+                    contributor__in=represented_users,
+                    textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
+                ).exists():
+                    return True
+
+                # the people responsible for a course can see all general text answers for all its evaluations
+                if textanswer.contribution.is_general and any(
+                    user in represented_users
+                    for user in textanswer.contribution.evaluation.course.responsibles.all()  # includes self
```

Comment:
```suggestion
                    user in represented_users  # includes self
                    for user in textanswer.contribution.evaluation.course.responsibles.all()
```
did you mean it like this?